common/src/main/scala/org/apache/celeborn/common/util/PbSerDeUtils.scala

```diff
@@ -318,4 +316,65 @@ object PbSerDeUtils {
     }).asJava
   }
 
+  def fromPbAppDiskUsage(pbAppDiskUsage: PbAppDiskUsage): AppDiskUsage = {
+    AppDiskUsage(pbAppDiskUsage.getAppId, pbAppDiskUsage.getEstimatedUsage)
+  }
+
+  def toPbAppDiskUsage(appDiskUsage: AppDiskUsage): PbAppDiskUsage = {
+    PbAppDiskUsage.newBuilder()
+      .setAppId(appDiskUsage.appId)
+      .setEstimatedUsage(appDiskUsage.estimatedUsage)
+      .build()
+  }
+
+  def fromPbAppDiskUsageSnapshot(
+      pbAppDiskUsageSnapShot: PbAppDiskUsageSnapshot): AppDiskUsageSnapShot = {
+    val snapShot = new AppDiskUsageSnapShot(pbAppDiskUsageSnapShot.getTopNItemsCount)
+    snapShot.startSnapShotTime = pbAppDiskUsageSnapShot.getStartSnapShotTime
+    snapShot.endSnapShotTime = pbAppDiskUsageSnapShot.getEndSnapshotTime
+    snapShot.topNItems =
+      pbAppDiskUsageSnapShot.getTopNItemsList.asScala.map(fromPbAppDiskUsage).toList.toArray
+    snapShot
+  }
+
+  def toPbAppDiskUsageSnapshot(snapshots: AppDiskUsageSnapShot): PbAppDiskUsageSnapshot = {
+    PbAppDiskUsageSnapshot.newBuilder()
+      .setStartSnapShotTime(snapshots.startSnapShotTime)
+      .setEndSnapshotTime(snapshots.endSnapShotTime)
+      .addAllTopNItems(snapshots.topNItems.map(toPbAppDiskUsage).toList.asJava)
+      .build()
+  }
+
+  def toPbSnapshotMetaInfo(
+      estimatedPartitionSize: java.lang.Long,
+      registeredShuffle: java.util.Set[String],
+      hostnameSet: java.util.Set[String],
+      blacklist: java.util.Set[WorkerInfo],
+      workerLostEvent: java.util.Set[WorkerInfo],
+      appHeartbeatTime: java.util.Map[String, java.lang.Long],
+      workers: java.util.ArrayList[WorkerInfo],
+      partitionTotalWritten: java.lang.Long,
+      partitionTotalFileCount: java.lang.Long,
+      appDiskUsageMetricSnapshots: Array[AppDiskUsageSnapShot],
+      currentAppDiskUsageMetricsSnapshot: AppDiskUsageSnapShot): PbSnapshotMetaInfo = {
+    val builder = PbSnapshotMetaInfo.newBuilder()
+      .setEstimatedPartitionSize(estimatedPartitionSize)
+      .addAllRegisteredShuffle(registeredShuffle)
+      .addAllHostnameSet(hostnameSet)
+      .addAllBlacklist(blacklist.asScala.map(toPbWorkerInfo(_, true)).asJava)
+      .addAllWorkerLostEvents(workerLostEvent.asScala.map(toPbWorkerInfo(_, true)).asJava)
+      .putAllAppHeartbeatTime(appHeartbeatTime)
+      .addAllWorkers(workers.asScala.map(toPbWorkerInfo(_, true)).asJava)
+      .setPartitionTotalWritten(partitionTotalWritten)
+      .setPartitionTotalFileCount(partitionTotalFileCount)
+      // appDiskUsageMetricSnapshots can have null values,
+      // protobuf repeated value can't support null value in list.
+      .addAllAppDiskUsageMetricSnapshots(appDiskUsageMetricSnapshots.filter(_ != null)
```

Comment:
Any good suggestions for this part? @waitinfuture @pan3793

Reply:
I think it's fine to write this way. In fact topNItems may also contain null, so we should modify toPbAppDiskUsageSnapshot too.